### ISSUE TITLE
Error in document creation causes cascade of errors leading to crash

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#5172]
An unhandled exception in document creation doesn't just break the document, it leaves things in a broken state, and closing the tab causes a hard crash

```
ERROR [2018-06-22 13:19:46Z]: 
System.ArgumentNullException: Value cannot be null.
Parameter name: definition
  at MonoDevelop.Ide.Editor.Highlighting.SyntaxHighlighting..ctor (MonoDevelop.Ide.Editor.Highlighting.SyntaxHighlightingDefinition definition, MonoDevelop.Ide.Editor.IReadonlyTextDocument document) [0x00056] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Highlighting/SyntaxHighlighting.cs:38 
  at MonoDevelop.MSBuildEditor.MSBuildTextEditorExtension.CheckHighlighting () [0x00052] in <f70eaf2c3bba4e8c866c158ab2865fab>:0 
  at MonoDevelop.MSBuildEditor.MSBuildTextEditorExtension.Initialize () [0x0003b] in <f70eaf2c3bba4e8c866c158ab2865fab>:0 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.Initialize (MonoDevelop.Ide.Editor.TextEditor editor, MonoDevelop.Ide.Editor.DocumentContext context) [0x0004f] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:65 
  at MonoDevelop.Ide.Editor.TextEditor.SetExtensionChain (MonoDevelop.Ide.Editor.DocumentContext documentContext, System.Collections.Generic.IEnumerable`1[T] extensions) [0x00077] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs:1190 
  at MonoDevelop.Ide.Editor.TextEditor.InitializeExtensionChain (MonoDevelop.Ide.Editor.DocumentContext documentContext) [0x00127] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs:1171 
  at MonoDevelop.Ide.Gui.Document.InitializeExtensionChain () [0x00001] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs:672 
  at MonoDevelop.Ide.Gui.Document.InitializeEditor () [0x00049] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs:703 
  at MonoDevelop.Ide.Gui.Document.OnDocumentAttached () [0x0000f] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs:709 
  at MonoDevelop.Ide.Gui.Workbench.WrapDocument (MonoDevelop.Ide.Gui.IWorkbenchWindow window) [0x00068] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:875 
  at MonoDevelop.Ide.Gui.ViewCommandHandlers..ctor (MonoDevelop.Ide.Gui.IWorkbenchWindow window) [0x0000f] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ViewCommandHandlers.cs:50 
  at MonoDevelop.Ide.Gui.SdiWorkspaceWindow.CreateCommandHandler () [0x00001] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs:122 
  at MonoDevelop.Ide.Gui.DefaultWorkbench.ShowView (MonoDevelop.Ide.Gui.ViewContent content, System.Boolean bringToFront, MonoDevelop.Ide.Gui.IViewDisplayBinding binding, MonoDevelop.Components.DockNotebook.DockNotebook notebook) [0x0014e] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs:445 
  at MonoDevelop.Ide.Gui.LoadFileWrapper+<Invoke>d__8.MoveNext () [0x00367] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1614 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:357 
  at MonoDevelop.Ide.Gui.Workbench+<RealOpenFile>d__118.MoveNext () [0x00491] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1094 
ERROR [2018-06-22 13:19:56Z]: Exception in UserPreferencesRequested.
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Gui.Workbench.AddNotebookDocuments (MonoDevelop.Ide.UserPreferencesEventArgs args, System.Collections.Generic.List`1[T] files, MonoDevelop.Components.DockNotebook.DockNotebook notebook, System.Int32 notebookId) [0x00024] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1179 
  at MonoDevelop.Ide.Gui.Workbench.OnStoringWorkspaceUserPreferences (System.Object s, MonoDevelop.Ide.UserPreferencesEventArgs args) [0x00130] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1149 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Ide.UserPreferencesEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Ide.UserPreferencesEventArgs)
  at MonoDevelop.Ide.RootWorkspace.SavePreferences (MonoDevelop.Projects.WorkspaceItem item) [0x0005d] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs:762 
ERROR [2018-06-22 13:20:06Z]: Exception in UserPreferencesRequested.
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Gui.Workbench.AddNotebookDocuments (MonoDevelop.Ide.UserPreferencesEventArgs args, System.Collections.Generic.List`1[T] files, MonoDevelop.Components.DockNotebook.DockNotebook notebook, System.Int32 notebookId) [0x00024] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1179 
  at MonoDevelop.Ide.Gui.Workbench.OnStoringWorkspaceUserPreferences (System.Object s, MonoDevelop.Ide.UserPreferencesEventArgs args) [0x00130] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1149 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Ide.UserPreferencesEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Ide.UserPreferencesEventArgs)
  at MonoDevelop.Ide.RootWorkspace.SavePreferences (MonoDevelop.Projects.WorkspaceItem item) [0x0005d] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs:762 
INFO [2018-06-22 13:22:19Z]: Add-in loaded: MonoDevelop.ILAsmBinding
objc[61366]: Class FIFinderSyncExtensionHost is implemented in both /System/Library/PrivateFrameworks/FinderKit.framework/Versions/A/FinderKit (0x7fffa73f9c90) and /System/Library/PrivateFrameworks/FileProvider.framework/OverrideBundles/FinderSyncCollaborationFileProviderOverride.bundle/Contents/MacOS/FinderSyncCollaborationFileProviderOverride (0x12273fcd8). One of the two will be used. Which one is undefined.
ERROR [2018-06-22 13:22:38Z]: Exception in UserPreferencesRequested.
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Gui.Workbench.AddNotebookDocuments (MonoDevelop.Ide.UserPreferencesEventArgs args, System.Collections.Generic.List`1[T] files, MonoDevelop.Components.DockNotebook.DockNotebook notebook, System.Int32 notebookId) [0x00024] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1179 
  at MonoDevelop.Ide.Gui.Workbench.OnStoringWorkspaceUserPreferences (System.Object s, MonoDevelop.Ide.UserPreferencesEventArgs args) [0x00130] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1149 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Ide.UserPreferencesEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Ide.UserPreferencesEventArgs)
  at MonoDevelop.Ide.RootWorkspace.SavePreferences (MonoDevelop.Projects.WorkspaceItem item) [0x0005d] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs:762 
ERROR [2018-06-22 13:22:52Z]: Exception in UserPreferencesRequested.
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Gui.Workbench.AddNotebookDocuments (MonoDevelop.Ide.UserPreferencesEventArgs args, System.Collections.Generic.List`1[T] files, MonoDevelop.Components.DockNotebook.DockNotebook notebook, System.Int32 notebookId) [0x00024] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1179 
  at MonoDevelop.Ide.Gui.Workbench.OnStoringWorkspaceUserPreferences (System.Object s, MonoDevelop.Ide.UserPreferencesEventArgs args) [0x00130] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1149 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Ide.UserPreferencesEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Ide.UserPreferencesEventArgs)
  at MonoDevelop.Ide.RootWorkspace.SavePreferences (MonoDevelop.Projects.WorkspaceItem item) [0x0005d] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs:762 
FATAL ERROR [2018-06-22 13:25:35Z]: An unhandled exception has occurred. Terminating MonoDevelop? True
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Gui.DefaultWorkbench.CloseClicked (System.Object o, MonoDevelop.Components.DockNotebook.TabEventArgs e) [0x00001] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs:1346 
  at MonoDevelop.Components.DockNotebook.DockNotebook.OnCloseTab (MonoDevelop.Components.DockNotebook.DockNotebookTab tab) [0x0000e] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.DockNotebook/DockNotebook.cs:411 
  at MonoDevelop.Components.DockNotebook.TabStrip.OnButtonReleaseEvent (Gdk.EventButton evnt) [0x00081] in /Users/mikayla/code/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.DockNotebook/TabStrip.cs:610 
  at Gtk.Widget.buttonreleaseevent_cb (System.IntPtr widget, System.IntPtr evnt) [0x00016] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/gtk-sharp-None/gtk/generated/Widget.cs:2267 
```

> VS bug [#638276](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/638276)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.344 (master)

Pull Request #5191 merged by: Mike Krüger
Author: mono
Commit: f21fe97b4de06b34170cfbd0d31e9a8aad229f95 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5191 merged by: Mike Kr??ger
Author: mono
Commit: f21fe97b4de06b34170cfbd0d31e9a8aad229f95 (xamarin/monodevelop)
